docs/architecture/sizing-guide.md

```diff
@@ -0,0 +1,337 @@
+at roughly 1Kb per request.
+
+### Disk
+lakeFS greatly benefits from fast local disks. 
+A lakeFS instance doesn't require any strong durability guarantees from the underlying storage, 
+as the disk is only ever used as a local caching layer for lakeFS metadata, and not for long-term storage.
+lakeFS is designed to work with [ephemeral disks](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ssd-instance-store.html){: target="_blank" } - 
+these are usually based on NVMe and are tied to the machine's lifecycle. 
+Using ephemeral disks lakeFS can provide a very high throughput/cost ratio, 
+probably the best that could be achieved on a public cloud, so we recommend those.
+
+a local cache of at least 512 MiB should be provided. 
+For large installations (managing >100 concurrently active branches, with >100M objects per commit),
+we recommend allocating at least 25 GiB - since it's a caching layer over a relatively slow storage (the object store), 
+the more the better - see [Important metrics](#important-metrics) below to understand how to do proper sizing.
+
+### PostgreSQL database
+
+lakeFS uses a PostgreSQL instance to manage branch references, authentication and authorization information 
+and to keep track of currently uncommitted data across branches.
+
+#### Storage
+The dataset stored in PostgreSQL is relatively modest, 
+as most metadata is pushed down into the object store. 
+Required storage is mostly a factor of the amount of uncommitted writes across all branches at any given point in time: 
+in the range of 150 MiB per every 100,000 uncommitted writes. 
+
+We recommend starting at 10 GiB for a production deployment, as it will likely be more than enough.
+
+#### RAM and `shared_buffers`
+Since the data size is small, it is recommended to provide enough memory to hold the vast majority of that data in RAM:
+Ideally configure [shared_buffers](https://www.postgresql.org/docs/current/runtime-config-resource.html#GUC-SHARED-BUFFERS){: target="_blank" } 
+of your PostgreSQL instances to be large enough to contain the currently active dataset. 
+Pick a database instance with enough RAM to accommodate this buffer size, at roughly x4 the size given for `shared_buffers` 
+(so for example, if an installation has ~500,000 uncommitted writes at any given time, it would require about 750 MiB of `shared_buffers`, 
+that would require about 3 GiB of RAM. 
+
+On AWS RDS, the default PostgreSQL parameter groups configure `shared_buffers` to be 25% of the chosen instance's available RAM, 
+so RDS users do not need to manually configure this.
+
+#### CPU
+
+PostgreSQL CPU cores help scale concurrent requests. Ideally 1 CPU core for every 5,000 requests/second is ideal.
+
+
+## Scaling factors
+
+Scaling lakeFS, like most data systems, moves across 2 axis: 
+concurrency (throughput of requests at a given timeframe), and latency (time to complete a single request).
```

Comment:
Not sure of this usage of "concurrency"; I usually don't use it as a measure of throughput.